pkg/apiserver/handlers.go

```diff
@@ -340,22 +333,20 @@ func (r *APIRequestInfoResolver) GetAPIRequestInfo(req *http.Request) (APIReques
 
 	// URL forms: /namespaces/{namespace}/{kind}/*, where parts are adjusted to be relative to kind
 	if currentParts[0] == "namespaces" {
-		if len(currentParts) < 3 {
-			requestInfo.Resource = "namespaces"
-			if len(currentParts) > 1 {
-				requestInfo.Namespace = currentParts[1]
-			}
-		} else {
-			requestInfo.Resource = currentParts[2]
+		if len(currentParts) > 1 {
 			requestInfo.Namespace = currentParts[1]
-			currentParts = currentParts[2:]
+
+			// if there is another step after the namespace name and it is not a known namespace subresource
+			// move currentParts to include it as a resource in its own right
+			if len(currentParts) > 2 {
```

Comment:
> if a path is just namespace/blah you want this to mean "object blah of type namespace in namespace blah"?

Yes.



> Doesn't that preclude useful behavior like listing all the resource types in a namespace?

I missed the original discussion on this (Note that this isn't changing behavior, simply making existing behavior a little more clear), but I the kubernetes `Authorizer` interface isn't really built to handle containment cleanly.  This is not a comment on the expression of policy, but on the `Authorizer` API itself.  It is built around a request operating against a single resource, not a resource that contains other resources.

Because of that, expressing an `Authorizer` that allows `ReadOnly` access to a `namespace` resource would be unable to reasonably apply rules to the contained resources.  Since the `Authorizer` operates as a single filter on the request, it would not have the opportunity to authorize access to the contained resources.

While agree that having a way to authorize requests against contained resources would be interesting, I don't think I want to change the existing behavior of `authorizer.AttributesRecord` here.